src/elf.rs

```diff
@@ -384,7 +396,24 @@ impl<'a, E: UserDefinedError, I: InstructionMeter> EBpfElf<E, I> {
                 }
                 None
             })
-            .collect();
+            .collect::<Vec<_>>();
+        ro_section_infos.sort_by(|a, b| a.vaddr.cmp(&b.vaddr));
+        for i in 0..ro_section_infos.len() {
+            if ro_section_infos[i].vaddr > ebpf::MM_STACK_START {
```

Comment:
Would a final check of all regions (or on each region add) for overlapping be more robust?

Reply:
Yes, I also considered that, was just a bit more involved due to introducing a `Result` type in the constructor of `MemoryRegion`. Anyway, I moved the overlap checks there.